{
  Future future=streamingSlopResults.submit(call);
  try {
    future.get();
  }
 catch (  InterruptedException e1) {
    logger.error("Callback failed",e1);
    throw new VoldemortException("Callback failed");
  }
catch (  ExecutionException e1) {
    logger.error("Callback failed during execution",e1);
    throw new VoldemortException("Callback failed during execution");
  }
}
